File: 1899-count-items-matching-a-rule/solution.cs
public class Solution {
    public int CountMatches(IList<IList<string>> items, string ruleKey, string ruleValue) {
        int count = 0;
        int index = 0;
        if (ruleKey == "color"){
            index = 1;
        }
        else if(ruleKey == "type"){
            index = 0;
        }
        else{
            index = 2;
        }
        foreach(IList<string> list in items){
            if (list[index] == ruleValue){
                count +=1;
            }
        }
        return count;
    }
}

File: 1899-count-items-matching-a-rule/solution.java
import java.util.List;

class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int count = 0;
        for (List<String> item : items) {
            if (("type".equals(ruleKey) && item.get(0).equals(ruleValue)) ||
                ("color".equals(ruleKey) && item.get(1).equals(ruleValue)) ||
                ("name".equals(ruleKey) && item.get(2).equals(ruleValue))) {
                count++;
            }
        }
        return count;
    }
}

